crates/pathfinder/src/state/sync.rs

```diff
@@ -701,6 +719,103 @@ fn deploy_contract(
         .context("Inserting class hash into contracts table")
 }
 
+/// Downloads and inserts class definitions for any classes in the
+/// list which are not already present in the database.
+async fn download_verify_and_insert_missing_classes<
```

Comment:
I think this is the third variation of this functionality now. I don't want to tinker too much more in this PR, so will leave it as a future exercise to declutter this.